sam-app/feature-tests/vitest-step-definitions/GovUKMobileCognitoUserPool.steps.ts

```diff
@@ -1,74 +1,87 @@
+                                  "Fn::Split": [
+                                    "/",
+                                    {
+                                      Ref: "AWS::StackId",
+                                    },
+                                  ],
                                 },
-                                "user-pool",
-                                {
-                                    "Fn::Select": [
-                                        4,
-                                        {
-                                            "Fn::Split": [
-                                                "-",
-                                                {
-                                                    "Fn::Select": [
-                                                        2,
-                                                        {
-                                                            "Fn::Split": [
-                                                                "/",
-                                                                {
-                                                                    Ref: "AWS::StackId"
-                                                                }
-                                                            ]
-                                                        }
-                                                    ]
-                                                }
-                                            ]
-                                        }
-                                    ]
-                                }
-                            ]
-                        ],
-                    },
-                    DeletionProtection: "ACTIVE",
-                    UsernameAttributes: ["email"],
-                    Schema: [{
-                        "AttributeDataType": "String",
-                        "Name": "email",
-                        "Required": true,
-                    }],
-                    UserPoolTags: {
-                        "Environment": { "Ref": "Environment" },
-                        "Product": "GOV.UK",
-                        "System": "Authentication"
+                              ],
+                            },
+                          ],
+                        },
+                      ],
                     },
-                }))
-
-        })
-    })
-})
-
+                  ],
+                ],
+              },
+              DeletionProtection: "ACTIVE",
+              UsernameAttributes: ["email"],
+              Schema: [
+                {
+                  AttributeDataType: "String",
+                  Name: "email",
+                  Required: true,
+                },
+              ],
+              UserPoolTags: {
+                Environment: { Ref: "Environment" },
+                Product: "GOV.UK",
+                System: "Authentication",
```

Comment:
Can we remove this from the assertion if it's already being checked in the, "do all the resource's have the correct tags" step?

Reply:
We're also using `Match.objectEquals` which looks for an exact object match https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.assertions.Match.html

Reply:
Ah okay, so even where the value is `Tags` (such as [WAF](https://github.com/alphagov/govuk-mobile-backend/pull/15/files#diff-ca6305d2e3f3dc2a6beba3888f5c10538a6c8b8855655f16a112d30c7f7f3306)) we have to repeat assertions? We can't use something like `objectLike` (partial match)?